homeassistant/components/tedee/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Tedee sensor entities."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pytedee_async import TedeeLock
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import TedeeDescriptionEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class TedeeSensorEntityDescription(SensorEntityDescription):
+    """Describes Tedee sensor entity."""
+
+    value_fn: Callable[[TedeeLock], float | None]
+
+
+ENTITIES: tuple[TedeeSensorEntityDescription, ...] = (
+    TedeeSensorEntityDescription(
+        key="battery_sensor",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda lock: lock.battery_level,
+    ),
+    TedeeSensorEntityDescription(
+        key="pullspring_duration",
+        translation_key="pullspring_duration",
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        state_class=SensorStateClass.TOTAL,
```

Comment:
Why do we set state class total? Isn't this just a read only setting and should have state class measurement? There's no accumulation happening, right?

https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes